tests/typ/bugs/newline-mode.typ

```diff
@@ -22,3 +22,53 @@
 else {
   ("1", "2")
 }
+
+---
+#{
+  "hi 1"
+
+    .clusters()
+}
```

Comment:
I know that the test existed before, but would you mind rewriting the new test cases that don't strictly need an image to use `test(..)` assertions instead? I want new tests to make more use of direct assertions compared to reference images, as they are smaller and easier to compare/review.

```suggestion
#test({
  "hi 1"

    .clusters()
}, ("h", "i", " ", "1"))
```